MovingBlocks/Terasology
# fix(networking-timeout): graceful shutdown after failed connection

### Contains

Make graceful game close after Connection failed when try to connect to server.
was non-daemon thread for timeout, which stucks on WAITING state when game closing.

Bonus - timeout thread named now. new license header for file

### How to test

1. open Terasology
2. Go to `Join Game`
3. try to connect to some stopped servers.
4. close game. (ideally faster then 10 seconds)
5. check that game closes (game process exited)
